yugabyte: run-jepsen log file name fix

svarnau opened the pull request:
New test names have slashes, but log file name cannot contain slash.

svarnau committed:
yugabyte: run-jepsen log file name fix

New test names have slashes, but log file name cannot contain slash.

svarnau committed:
yugabyte/run-jepsen: rework logging

Remove timestamps from log file names.
Save stdout files if test fails.
Change format of end-of-run summary.

svarnau committed:
yugabyte/run-jepsen: add arguments for workloads and nemeses

svarnau committed:
yugabyte/run-jepsen: rework loggin again and set exit code

svarnau committed:
yugabyte/run-jepsen: fix syntax

svarnau commented:
Several changes piled up based on testing in sample jenkins job.

frozenspider commented on yugabyte/run-jepsen.py:
```diff
@@ -232,8 +231,16 @@ def parse_args():
     parser.add_argument(
         '--concurrency',
-        default='4n',
+        default='5n',
```
`single-key-acid` test uses `concurrent-generator`, which requires a concurrency to be a multiple of 2. I suggest keeping default concurrency at `4n`.

frozenspider commented on yugabyte/run-jepsen.py:
```diff
@@ -272,106 +279,99 @@ def main():
+            total_test_time_sec += test_elapsed_time_sec
+
+            total_elapsed_time_sec = time.time() - start_time
+            logging.info("Finished running %d tests.", num_tests_run)
+            logging.info("    %d OK, %d Problems (%d Timed-Out)",
+                num_everything_looks_good, num_not_everything_looks_good, num_timed_out_tests)
+            logging.info("    %d exited success, %d exited failure.", num_zero_exit_code,
+                num_non_zero_exit_code)
+            logging.info("Elapsed time: %.1f sec, Test time: %.1f sec, Avg test time: %.1f sec", 
+                total_elapsed_time_sec, total_test_time_sec,
```
Weird uppercases (after a comma or a digit), keep them consistent?
Also, is this condensed output still easily readable?

frozenspider commented on yugabyte/run-jepsen.py:
```diff
@@ -272,106 +279,99 @@ def main():
+            logging.info("Finished running %d tests.", num_tests_run)
+            logging.info("    %d OK, %d Problems (%d Timed-Out)",
+                num_everything_looks_good, num_not_everything_looks_good, num_timed_out_tests)
+            logging.info("    %d exited success, %d exited failure.", num_zero_exit_code,
```
Nit: maybe something like `"%d test out of %d returned non-zero exit code", num_non_zero_exit_code, num_tests_run`?

frozenspider reviewed (commented):
Reviewed, left a few minor comments

svarnau committed:
Logging format per review, allow multiple nemesis per test run.

Change nemesis option to be space separated, with groups to be used
together as comma-seperated.

svarnau committed:
Fix some missing quotes around path names.

frozenspider commented on yugabyte/run-jepsen.py:
```diff
@@ -234,6 +230,14 @@ def parse_args():
+    parser.add_argument(
+        '--nemeses',
+        default=' '.join(NEMESES),
+        help='Comma-seperated list of nemeses. Default: ' + ' '.join(NEMESES))
```
This is weird - right now tooltip says:
```
  --nemeses NEMESES     Comma-seperated list of nemeses. Default: none kill-
                        tserver kill-master pause-tserver pause-master
                        partition kill,pause,partition
```
So is it comma-separated or space-separated?

svarnau replied:
Good point. Was changed to accommodate a jenkins restriction, that no longer applies.

frozenspider reviewed (commented):
Left one more comment

svarnau committed:
Fix nemeses option to be comma-separated list.

frozenspider reviewed (approved):
LGTM!

frozenspider commented on yugabyte/run-jepsen.py:
```diff
@@ -249,7 +252,7 @@ def main():
 
     start_time = time.time()
-    nemeses = NEMESES
+    nemeses = args.nemeses.split(' ')
```
Actually, shouldn't this be `split(',')`?

svarnau replied:
Ah, this is the key issue. When do we want to run nemesis one at a time, and when do we want to pass multiple down to the to the test all together? That is the real reason I had space and comma separators.  That is really too subtle. The way jenkins is using this, we should just pass these down, not loop through them.

Instead we only loop through  the workloads.

svarnau committed:
Pass nemeses down to test, rather than loop through them.

svarnau committed:
Fix exit code.
